src/actions/etherspotActions.js

```diff
@@ -191,12 +199,44 @@ export const refreshEtherspotAccountsAction = () => {
     await Promise.all(
       etherspotAccounts.map(async ({ address: etherspotAccountAddress }) => {
         const extra = await etherspotService.getAccountPerChains(etherspotAccountAddress);
+        if (isEtherspotAccount(account)) handleAccountStateChange(extra, account?.extra, dispatch);
         dispatch(updateAccountExtraIfNeededAction(etherspotAccountAddress, extra));
       }),
     );
   };
 };
 
+const getChain = (chain: Chain) => {
+  switch (chain) {
+    case CHAIN_NAMES.ETHEREUM:
+      return CHAIN_SHORT.ETHEREUM.toLowerCase();
+    case CHAIN_NAMES.BINANCE:
+      return CHAIN_SHORT.BINANCE.toLowerCase();
+    case CHAIN_NAMES.XDAI:
+      return CHAIN_SHORT.XDAI.toLowerCase();
+    case CHAIN_NAMES.POLYGON:
+      return CHAIN_SHORT.POLYGON.toLowerCase();
+    default:
+      return '';
+  }
+};
```

Comment:
After seeing the definitions of `CHAIN_NAMES` and `CHAIN_SHORT`, it feels like we may have gone the long way around here. Where we have `const chains = Object.keys(etherspotAccountExtra)`, how does the data look? Can we not `toLowerCase()` this string in some way?